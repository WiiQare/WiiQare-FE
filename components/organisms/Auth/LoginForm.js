import React from "react";
import Link from "next/link";
import TextField from "@mui/material/TextField";
import Box from "@mui/material/Box";
import Stepper from "@mui/material/Stepper";
import Step from "@mui/material/Step";
import StepButton from "@mui/material/StepButton";
import Button from "@mui/material/Button";
import Stack from "@mui/material/Stack";
import useMediaQuery from "@mui/material/useMediaQuery";
import Visibility from "@mui/icons-material/Visibility";
import VisibilityOff from "@mui/icons-material/VisibilityOff";
import { signIn } from "next-auth/react"

import OnboardingScreen from "../../molecules/OnboardingScreen";
import { useLocalStorage } from "../../../hooks/useLocalStorage";
import {
	FormControl,
	IconButton,
	InputAdornment,
	InputLabel,
	OutlinedInput,
	Typography,
} from "@mui/material";
import MenuHolder from "../../atoms/MenuHolder";
import BlinkSnackbar from "../../atoms/BlinkSnackbar";
import Image from "next/image";

function LoginForm() {
	const matches = useMediaQuery("(max-width: 992px)");
	const [onboardingIsEnd, setOnboardingIsEnd] = React.useState("");
	const [showPassword, setShowPassword] = React.useState(false);

	const handleClickShowPassword = () => setShowPassword((show) => !show);

	React.useEffect(() => {
		setOnboardingIsEnd(localStorage.getItem("onboardingIsEnd") || "");
	}, []);

	//Handle Google Login
    const handleGoogleSignIn = async () => {
        //signIn('google', {callbackUrl: "http://localhost:3000"})
        signIn('google', {callbackUrl: "https://wiiqare-unicef-ii.vercel.app"})

    }

	const handleLinkedInSignIn = async () => {
        //signIn('linkedin', {callbackUrl: "http://localhost:3000"})
        signIn('linkedin', {callbackUrl: "https://wiiqare-unicef-ii.vercel.app"})
    }

	const formHolder = (
		<>
			<div className="form-holder">
				<MenuHolder href="/register" label="SIGN UP" />
				<div className="signin-signup-form">
					<div className="flex flex-col md:w-3/4 w-full border-opacity-50 gap-4">
						
						<div className="form-items !w-full !max-w-full flex flex-col items-center mx-auto">
							<div className="md:w-3/4 mx-auto">
								<div className="form-title">Sign in to your account</div>
								<Box sx={{ mb: 3, mt: 2 }}></Box>
								<form id="signinform" className="">
									<Stack spacing={2}>
										<TextField
											id="outlined-basic"
											fullWidth
											label="E-mail Address"
											variant="outlined"
										/>
										<FormControl fullWidth variant="outlined">
											<InputLabel htmlFor="outlined-basic1">Password</InputLabel>
											<OutlinedInput
												id="outlined-basic1"
												label="Password"
												type={showPassword ? "text" : "password"}
												endAdornment={
													<InputAdornment position="end">
														<IconButton
															aria-label="toggle password visibility"
															onClick={handleClickShowPassword}
															edge="end"
														>
															{showPassword ? <VisibilityOff /> : <Visibility />}
														</IconButton>
													</InputAdornment>
												}
											/>
										</FormControl>
										<Box sx={{ textAlign: "right", mb: 2 }}>
											<Typography color="text.secondary" variant="caption">
												<Link href={"/forgot/password"} legacyBehavior>
													<a>
														Forgot your password ?
													</a>
												</Link>
											</Typography>
										</Box>
										<Box>
											<Button
												size="large"
												variant="contained"
												onClick={() => { }}
												className="bg-yellow"
											>
												SIGN IN
											</Button>
										</Box>
									</Stack>
								</form>
							</div>
						</div>

						<div className="divider"><span className="text-gray-400">OR</span></div>

						<div className="flex flex-col md:flex-row gap-4 mb-2">
							<button type="button" className="w-full py-3 flex justify-center items-center gap-4 hover:bg-gray-200 rounded-xl border font-medium" onClick={handleGoogleSignIn}><Image src={"/images/google.svg"} width="20" height="20" alt="Google logo" /> Sign In with Google</button>
							<button type="button" className="w-full py-3 flex justify-center items-center gap-4 hover:bg-gray-200 rounded-xl border font-medium" onClick={handleLinkedInSignIn}><Image src={"/images/linkedin.png"} width="25" height="25" alt="Facebook logo" /> Sign In with LinkedIn</button>
						</div>
					</div>
				</div>
			</div>
			<BlinkSnackbar />
		</>
	);
	return (
		<>
			{matches && !onboardingIsEnd ? (
				<OnboardingScreen
					onStartClick={() => {
						if (typeof window !== "undefined") {
							localStorage.setItem("onboardingIsEnd", "onboardingIsEnd");
						}
						setOnboardingIsEnd("onboardingIsEnd");
					}}
				/>
			) : (
				formHolder
			)}
		</>
	);
}

export default LoginForm;
