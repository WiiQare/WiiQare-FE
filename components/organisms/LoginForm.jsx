import React from "react";
import Link from "next/link";
import TextField from "@mui/material/TextField";
import Box from "@mui/material/Box";
import Stepper from "@mui/material/Stepper";
import Step from "@mui/material/Step";
import StepButton from "@mui/material/StepButton";
import Button from "@mui/material/Button";
import Stack from "@mui/material/Stack";
import useMediaQuery from "@mui/material/useMediaQuery";
import Visibility from "@mui/icons-material/Visibility";
import VisibilityOff from "@mui/icons-material/VisibilityOff";

import OnboardingScreen from "../molecules/OnboardingScreen";
import { useLocalStorage } from "../../hooks/useLocalStorage";
import {
  FormControl,
  IconButton,
  InputAdornment,
  InputLabel,
  OutlinedInput,
  Typography,
} from "@mui/material";
import MenuHolder from "../atoms/MenuHolder";
import BlinkSnackbar from "../atoms/BlinkSnackbar";

function LoginForm() {
  const matches = useMediaQuery("(max-width: 992px)");
  const [onboardingIsEnd, setOnboardingIsEnd] = React.useState("");
  const [showPassword, setShowPassword] = React.useState(false);

  const handleClickShowPassword = () => setShowPassword((show) => !show);

  React.useEffect(() => {
    setOnboardingIsEnd(localStorage.getItem("onboardingIsEnd") || "");
  }, []);

  const formHolder = (
    <>
      <div className="form-holder">
        <MenuHolder href="/register" label="SIGN UP" />
        <div className="signin-signup-form">
          <div className="form-items">
            <div className="form-title">Sign in to your account</div>
            <Box sx={{ mb: 3, mt: 2 }}></Box>
            <form id="signinform">
              <Stack spacing={2}>
                <TextField
                  id="outlined-basic"
                  fullWidth
                  label="E-mail Address"
                  variant="outlined"
                />
                <FormControl fullWidth variant="outlined">
                  <InputLabel htmlFor="outlined-basic1">Password</InputLabel>
                  <OutlinedInput
                    id="outlined-basic1"
                    label="Password"
                    type={showPassword ? "text" : "password"}
                    endAdornment={
                      <InputAdornment position="end">
                        <IconButton
                          aria-label="toggle password visibility"
                          onClick={handleClickShowPassword}
                          edge="end"
                        >
                          {showPassword ? <VisibilityOff /> : <Visibility />}
                        </IconButton>
                      </InputAdornment>
                    }
                  />
                </FormControl>
                <Box sx={{ textAlign: "right", mb: 2 }}>
                  <Typography color="text.secondary" variant="caption">
                    Forgot your password ?
                  </Typography>
                </Box>
                <Box>
                  <Button
                    size="large"
                    variant="contained"
                    onClick={() => {}}
                    className="bg-yellow"
                  >
                    SIGN IN
                  </Button>
                </Box>
              </Stack>
            </form>
          </div>
        </div>
      </div>
      <BlinkSnackbar />
    </>
  );
  return (
    <>
      {matches && !onboardingIsEnd ? (
        <OnboardingScreen
          onStartClick={() => {
            if (typeof window !== "undefined") {
              localStorage.setItem("onboardingIsEnd", "onboardingIsEnd");
            }
            setOnboardingIsEnd("onboardingIsEnd");
          }}
        />
      ) : (
        formHolder
      )}
    </>
  );
}

export default LoginForm;
