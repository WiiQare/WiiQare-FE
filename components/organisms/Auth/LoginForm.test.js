import React from 'react';
import { render } from '@testing-library/react';
import LoginForm from './LoginForm';
import { QueryClientProvider, QueryClient } from 'react-query';

jest.mock('next/router', () => ({
  useRouter: jest.fn().mockReturnValue({
    router: [],
  }),
}));

describe('LoginForm', () => {
  it('renders', () => {
    const { container } = render(
      <QueryClientProvider client={new QueryClient()}>
        <LoginForm />
      </QueryClientProvider>,
    );
    expect(container).toMatchSnapshot();
  });
});
