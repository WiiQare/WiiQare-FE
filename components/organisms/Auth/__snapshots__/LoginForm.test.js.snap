// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm renders 1`] = `
<div>
  <div
    class="form-holder"
  >
    <div
      class="menu-holder"
    >
      <ul
        class="main-links"
      >
        <li>
          <div
            class="MuiBox-root css-1at62qq"
          >
            <a
              class="normal-link"
              href="/register"
            >
              Vous n'avez pas de compte ?
            </a>
          </div>
        </li>
        <li>
          <div
            class="MuiBox-root css-0"
          >
            <a
              href="/register"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium bg-yellow css-sghohy-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="button"
              >
                S'INSCRIRE
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </a>
          </div>
        </li>
      </ul>
    </div>
    <div
      class="signin-signup-form"
    >
      <div
        class="flex flex-col md:w-3/4 w-full border-opacity-50 gap-4"
      >
        <div
          class="form-items !w-full !max-w-full flex flex-col items-center mx-auto"
        >
          <div
            class="md:w-3/4 mx-auto"
          >
            <div
              class="form-title"
            >
              Connectez-vous à votre compte
            </div>
            <div
              class="MuiBox-root css-19bt0nb"
            />
            <form
              id="signinform"
            >
              <div
                class="css-1p5q5e5-MuiStack-root"
              >
                <div
                  class="space-y-1 flex flex-col gap-1"
                >
                  <div
                    class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root lowercase css-wb57ya-MuiFormControl-root-MuiTextField-root"
                  >
                    <label
                      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                      data-shrink="false"
                      for="outlined-basic"
                      id="outlined-basic-label"
                    >
                      Entrez votre adresse email ou Numéro de Téléphone
                    </label>
                    <div
                      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl input-email css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
                    >
                      <input
                        aria-invalid="false"
                        class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                        id="outlined-basic"
                        name="email"
                        type="text"
                        value=""
                      />
                      <fieldset
                        aria-hidden="true"
                        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                      >
                        <legend
                          class="css-yjsfm1"
                        >
                          <span>
                            Entrez votre adresse email ou Numéro de Téléphone
                          </span>
                        </legend>
                      </fieldset>
                    </div>
                  </div>
                </div>
                <div
                  class="space-y-1 flex flex-col gap-1"
                >
                  <div
                    class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
                  >
                    <label
                      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                      data-shrink="false"
                      for="outlined-basic1"
                    >
                      Entrez votre mot de passe
                    </label>
                    <div
                      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-o9k5xi-MuiInputBase-root-MuiOutlinedInput-root"
                    >
                      <input
                        aria-invalid="false"
                        class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
                        id="outlined-basic1"
                        name="password"
                        type="password"
                        value=""
                      />
                      <div
                        class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
                      >
                        <button
                          aria-label="toggle password visibility"
                          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
                          tabindex="0"
                          type="button"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                            data-testid="VisibilityIcon"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
                            />
                          </svg>
                          <span
                            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                          />
                        </button>
                      </div>
                      <fieldset
                        aria-hidden="true"
                        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                      >
                        <legend
                          class="css-yjsfm1"
                        >
                          <span>
                            Entrez votre mot de passe
                          </span>
                        </legend>
                      </fieldset>
                    </div>
                  </div>
                </div>
                <div
                  class="MuiBox-root css-xtaagk"
                >
                  <span
                    class="MuiTypography-root MuiTypography-caption css-ol7wxs-MuiTypography-root"
                  >
                    <a
                      href="/forgot/password"
                    >
                      Mot de passe oublié ?
                    </a>
                  </span>
                </div>
                <div
                  class="MuiBox-root css-0"
                >
                  <button
                    class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge bg-yellow text-uppercase login css-jh47zj-MuiButtonBase-root-MuiButton-root"
                    tabindex="0"
                    type="submit"
                  >
                    Se connecter
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                </div>
              </div>
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
